src/components/FilterModal/FilterModal.jsx

```diff
@@ -1,25 +1,51 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { Button } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
 import Modal from '../Modal/Modal';
 import CheckBoxList from '../CheckBoxList/CheckBoxList';
+import { getFilterSetter, getFilterResetter } from '../../filters/freaksFilter';
 
-function FilterModal({ title, isOpen, onClose, keywords }) {
+function FilterModal({ title, isOpen, onClose, keywords, setOpenModal }) {
   const [ checkedState, setCheckedState ] = useState([]);
+  const queuedFilters = useRef([]);
+  const dispatch = useDispatch();
+
+  const onChangeCb = (result) => {
+    setCheckedState(result);
+    queuedFilters.current = [ ...result ];
```

Comment:
No need to spread the array

```suggestion
    queuedFilters.current = result;
```